GrammarNazi.App/HostedServices/TwitterBotHostedService.cs

```diff
@@ -112,21 +112,41 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 
-                        // TODO: Separate corrections in various tweets when max characters exceeded https://github.com/nminaya/grammar-nazi-bot/issues/73
-                        if (correctionString.Length - mentionedUsers.Length > Defaults.TwitterTextMaxLength)
-                            continue;
-
                         _logger.LogInformation($"Sending reply to: {tweet.CreatedBy.ScreenName}");
-                        var publishTweetParameters = new PublishTweetParameters(correctionString)
-                        {
-                            InReplyToTweetId = tweet.Id
-                        };
-                        var replyTweet = await _twitterClient.Tweets.PublishTweetAsync(publishTweetParameters);
 
-                        if (replyTweet != null)
+                        if (correctionString.Length < Defaults.TwitterTextMaxLength)
+                        {
+                            var publishTweetParameters = new PublishTweetParameters(correctionString)
+                            {
+                                InReplyToTweetId = tweet.Id
+                            };
+                            var replyTweet = await _twitterClient.Tweets.PublishTweetAsync(publishTweetParameters);
+
+                            if (replyTweet != null)
+                            {
+                                _logger.LogInformation("Reply sent successfuly");
+                                await _twitterLogService.LogReply(tweet.Id, replyTweet.Id);
+                            }
+                        }
+                        else
                         {
-                            _logger.LogInformation("Reply sent successfuly");
-                            await _twitterLogService.LogReply(tweet.Id, replyTweet.Id);
+                            var replyTweets = correctionString.SplitInParts(Defaults.TwitterTextMaxLength);
+
+                            foreach (var reply in replyTweets)
+                            {
+                                var correctionStringSplitted = $"@{tweet.CreatedBy.ScreenName} {mentionedUsers} {reply}";
+                                var publishTweetsParameters = new PublishTweetParameters(correctionStringSplitted)
+                                {
+                                    InReplyToTweetId = tweet.Id
+                                };
+                                var replyTweetSplitted = await _twitterClient.Tweets.PublishTweetAsync(publishTweetsParameters);
+
+                                if (replyTweetSplitted != null)
+                                {
+                                    _logger.LogInformation("Reply sent successfuly");
+                                    await _twitterLogService.LogReply(tweet.Id, replyTweetSplitted.Id);
+                                }
+                            }
                         }
 
                         await Task.Delay(_twitterBotSettings.PublishTweetDelayMilliseconds);
```

Comment:
@Jadhielv  I think we should copy this line into the foreach above after publishing the tweet in line 142. So that we avoid Twitter Limits...

Reply:
Done @nminaya